### DISCUSSION TITLE
Where can I find that how implemented videos and images viewer?

### DISCUSSION THREAD

**Discussion by GeeseBumps** [fluttercandies/flutter_wechat_assets_picker#90] [Q&A] [answered]
Hi, first of all, I'm really appreciated for using this great package.

I used to build my own custom image and video picker with photo_manager.

I played the video which I picked with VideoPlayer package, however there was quite latency if video file size is big.

However, your package provides really quick response no matter how the video file size is big! (And also image load is really quick when I click selected images!)

I want to know how you built this! I digged source codes for several hours but I couldn't find the part that coded video play and image viewer. 

Thank you!

**Comment by AlexV525** [comment, marked as answer]
Thanks for your compliment! The video builder can be found at here: https://github.com/fluttercandies/flutter_wechat_assets_picker/blob/b54b32fbc6ea743eb41d9c794d0a16e0bf6b5852/lib/src/widget/builder/video_page_builder.dart
